### ISSUE TITLE
Clean incomplete because device or resource is busy

### ISSUE THREAD

**Issue by ghost** [clearkimura/Customizer#82]
This is the other issue I encounter since I started using devel branch (also not sure since which commit). But may be it is just my side, or 14.04 host system issue. Screenshots as below.

Case 1: Most of the time, the **Clean** button failed to do cleaning:
![screenshot - 10122014 - 033836 am](https://cloud.githubusercontent.com/assets/1223304/5049931/a58d84e6-6c64-11e4-8b9b-9beba424d1fc.png)

When facing case 1, there will be an error dialog showing:

```
 [Errno 16] Device or resource busy: '/home/FileSystem/tmp'
```

Even after closing the error dialog, GUI still showing as if ISO is still loaded (username and hostname fields and buttons are active, but clicking any of these wouldn't run anything since most of files have been cleaned up in fact).

At this stage, cleaning using command line `sudo customizer -t` didn't work. Screenshot below showing I attempted to manually removing the files using `rm` which also failed to do the job.

![screenshot - 10122014 - 034029 am](https://cloud.githubusercontent.com/assets/1223304/5050510/bde380fe-6c69-11e4-976b-d19ba3a851eb.png)

As workaround, host system need to reboot. Even so, GUI cannot be run at all afterwards, since cleaning was incomplete. Screenshot below shows the error when attempting to run `sudo customizer-gui` immediately after reboot.

![screenshot - 10122014 - 034100 am](https://cloud.githubusercontent.com/assets/1223304/5050517/cb4f57cc-6c69-11e4-8fc0-9b37ee863fd6.png)

Finally, I had to re-run cleaning command `sudo customizer -t` via Terminal to remove leftover files which had failed to be cleaned in earlier session.

Everything works after reboot and redo clean.

Case 2: Cleaning was successful but Xubuntu crash report dialog appears:
![screenshot - 11152014 - 011406 am](https://cloud.githubusercontent.com/assets/1223304/5050128/822a8fa6-6c66-11e4-87d8-b100ed8c6fb9.png)

The crash dialog details cannot be exported to text file whenever it appears. From my brief reading, `gvfsd-trash` is said to have crashed and associated packages which are outdated caused this to happen.

I suspect the crash is may be because I had solely upgraded my host system kernels (and ignore other essential packages from being updated) while investigating issue #57 last month.

Case 3: Another attempt shows cleaning was successful without problem:
![screenshot - 11152014 - 012737 am](https://cloud.githubusercontent.com/assets/1223304/5050139/9f609dcc-6c66-11e4-981e-d7714dcf8bb4.png)

Above all, I won't jump to conclusion yet since this issue is somewhat happening at random. But I thought it is better to submit as issue, should there be anyone facing similar issue.

I will update other packages and monitor if things improved.

P.S.: Updating all essential packages didn't help, after all.

**Comment by fluxer** [comment]
There are several issues here which lead to this.

Worst of all the bind mounted filesystems are not unmounted before removing the content of the working directories. This is a major issue and should never happen, the fix would be easy and I will look into it tonight.

String escaping issues I have faced when Python version 3 is in use. Python version 2 shebang is set in all scripts but if you run it with something like `gksu python main.py` or `gksu python gui.py` and /usr/bin/python is symlink, hardlink or even a actual file that points to the Python 3 interpreter this can happen. And since I have not (yet) tested the code with Python 3 much a check and bail out when someone attempts to run it with something other than Python 2 would make sense until Python 3 compatibility issues are resolved. That is something I will look into tonight too.

The subprocess issue should be fixed. The casper.conf issues should be fixed by f1356e1, did you use fetch the latest version when you encountered this problem?

**Comment by fluxer** [comment]
I presume the issue has been addressed, awaiting confirm.

**Comment by ghost** [comment]
@fluxer 

> The casper.conf issues should be fixed by f1356e1, did you use fetch the latest version when you encountered this problem?

Yes, reproduced with same condition as preceding issue; using Customizer devel branch (latest commit 92e4b60) on 32-bit host system.

> I presume the issue has been addressed, awaiting confirm.

For the latter, I have installed latest devel branch commit ea5d850 to confirm. I have selected and loaded various 12.04 and 14.04 ISO images for 5 consecutive times. All can do cleaning properly.

The fix is success, well done!
